File: second-hand-trade-backend/src/main/java/com/campus/yujianhaowu/model/entity/CulturalContent.java
package com.campus.yujianhaowu.model.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("cultural_contents")
public class CulturalContent extends BaseEntity {

    @TableField("title")
    private String title;

    @TableField("summary")
    private String summary;

    @TableField("content")
    private String content;

    @TableField("cover_image")
    private String coverImage;

    @TableField("category")
    private String category;

    @TableField("tags")
    private String tags;

    @TableField("author_id")
    private Long authorId;

    @TableField("view_count")
    private Integer viewCount;

    @TableField("like_count")
    private Integer likeCount;

    @TableField("comment_count")
    private Integer commentCount;

    @TableField("favorite_count")
    private Integer favoriteCount;

    @TableField("is_top")
    private Boolean isTop;

    @TableField("is_recommend")
    private Boolean isRecommend;

    @TableField("is_published")
    private Boolean isPublished;

    @TableField("published_at")
    private LocalDateTime publishedAt;
}

File: second-hand-trade-frontend/src/api/modules/culturalContent.ts
import request from '@/api'
import type { ApiResponse, PageResult } from '@/types'

export interface CulturalContent {
  id: number
  title: string
  summary: string
  content: string
  coverImage: string
  category: string
  tags: string[]
  authorId: number
  authorName: string
  viewCount: number
  likeCount: number
  commentCount: number
  favoriteCount: number
  isTop: boolean
  isRecommend: boolean
  isPublished: boolean
  publishedAt: string
  createdAt: string
  updatedAt: string
}

export interface CulturalContentCreateParams {
  title: string
  summary?: string
  content: string
  coverImage?: string
  category?: string
  tags?: string[]
  isPublished?: boolean
}

/**
 * 获取文化资讯列表
 */
export function getCulturalContents(params: {
  category?: string
  keyword?: string
  current: number
  size: number
}) {
  return request<ApiResponse<PageResult<CulturalContent>>>({
    url: '/cultural-contents',
    method: 'get',
    params,
  })
}

/**
 * 获取推荐资讯
 */
export function getRecommendContents(params: {
  current: number
  size: number
}) {
  return request<ApiResponse<PageResult<CulturalContent>>>({
    url: '/cultural-contents/recommend',
    method: 'get',
    params,
  })
}

/**
 * 获取资讯详情
 */
export function getContentDetail(id: number) {
  return request<ApiResponse<CulturalContent>>({
    url: `/cultural-contents/${id}`,
    method: 'get',
  })
}

/**
 * 创建资讯
 */
export function createContent(data: CulturalContentCreateParams) {
  return request<ApiResponse<CulturalContent>>({
    url: '/cultural-contents',
    method: 'post',
    data,
  })
}

/**
 * 更新资讯
 */
export function updateContent(id: number, data: CulturalContentCreateParams) {
  return request<ApiResponse<void>>({
    url: `/cultural-contents/${id}`,
    method: 'put',
    data,
  })
}

/**
 * 删除资讯
 */
export function deleteContent(id: number) {
  return request<ApiResponse<void>>({
    url: `/cultural-contents/${id}`,
    method: 'delete',
  })
}
